main.py

```diff
@@ -27,19 +31,90 @@
 #       - [ ] The filepath to write the output of the solved function to.
 
 
-@click.command()
-@click.option('--solver', default="cello", help='Which Solver to Use.')
-def main_entry(solver: str, filepath: str, output_directory: str):
-    '''
+__version__ = "0.1.0"
+valid_solvers = [
+    ("Cello", "Built-in Cello module which predicts how well its circuits are likely to perform."),
+    ("Sleight", "Roughly evaluates the evolutionary stability of a circuit."),
+    ("Qian", "Calculates a intracellular resource demand coefficient for a circuit."),
+    ("Cummins", "Uses the DSGRN Database to summarize network dynamics of a circuit."),
+    ("Beal", "Characterizes the efficacy of a circuit based on signal-to-noise ratio."),
+    ("Gedeon", "Ranks a circuit by how robustly it supports hysteresis.")
+]
+
+def name_callback(value: str):
+    """
+    Checks whether the first input argument is a valid solver.
+    """
+    if value.title() not in [x[0] for x in valid_solvers]:
```

Comment:
Generally, when you want to normalize text you either pick Upper or Lower. Titlecase can be non-deterministic. I get that you wanted to match a visually pretty entry above.

I think what you're trying to do here is to do some sort of exclusionary testing. (Make sure the input is in my list of valid inputs). I'm going to make a little commit after this to show you the canonical-ish way of doing things, but feel free to push back if you feel your original intention was better.

Reply:
Dictionaries save the day! Thanks-- calling dict keys makes exclusionary testing much smoother.